core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -313,6 +313,17 @@ void CudaInternal::initialize(cudaStream_t stream) {
     (void)scratch_space(reduce_block_count * 16 * sizeof(size_type));
   }
 
+  // Allocate a staging buffer for constant mem in pinned host memory
+  // and an event to avoid overwriting driver for previous kernel launches
+  if (!constantMemHostStagingPerDevice[m_cudaDev])
+    KOKKOS_IMPL_CUDA_SAFE_CALL((cuda_malloc_host_wrapper(
+        reinterpret_cast<void **>(&constantMemHostStagingPerDevice[m_cudaDev]),
+        CudaTraits::ConstantMemoryUsage)));
+
+  if (!constantMemReusablePerDevice[m_cudaDev])
+    KOKKOS_IMPL_CUDA_SAFE_CALL(
+        (cuda_event_create_wrapper(&constantMemReusablePerDevice[m_cudaDev])));
```

Comment:
Move this block you are adding right after the `CudaInternal::cuda_devices.insert(m_cudaDev);`

Reply:
Sure.